python/cudf/cudf/_lib/parquet.pyx

```diff
@@ -878,14 +882,39 @@ cdef class ParquetReader:
         return df
 
     def read(self):
-        dfs = []
+        dfs = self._read_chunk()
         while self._has_next():
-            dfs.append(self._read_chunk())
-        df = cudf.concat(dfs)
-        df = _process_metadata(df, self.result_meta, self.names, self.row_groups,
-                               self.filepaths_or_buffers, self.pa_buffers,
-                               self.allow_range_index, self.cpp_use_pandas_metadata)
-        return df
+            dfs = [dfs, self._read_chunk()]
+            concatenated_columns = []
+            column_names = dfs[0]._column_names
+            all_columns = []
+            for i in range(len(column_names)):
+                cols = [table._data.columns[i] for table in dfs]
+                all_columns.append(cols)
+            i = 0
+            del dfs
+            while True:
+                try:
+                    columns = all_columns[0]
+                except IndexError:
+                    break
+                if len(columns) == 0:
+                    break
+                res_col = None
+                res_col = concat_columns(columns)
+                del all_columns[0]
+                concatenated_columns.append(res_col.to_pylibcudf(mode="read"))
+            dfs = cudf.DataFrame._from_data(
```

Comment:
Can this final dataframe still be made at the end? e.g.

```python
chunk = self._read_chunk()
column_names = chunk._column_names
concatenated_columns = list(chunk._columns)

while self._has_next():
    chunk = self._read_chunk()
    concatenated_columns = [concat_columns([prev_chunk, curr_chunk]) for prev_chunk, curr_chunk in zip(concatenated_columns, chunk._columns)]

dfs = cudf.DataFrame._from_data(dict(zip(column_names, concatenated_columns))
```

Reply:
Simplified the approach to free up the memory on a rolling basis as concat's happen.